### ISSUE TITLE
On mobile, the database never gets set to the onReady state

### ISSUE THREAD

**Issue by balmas** [alpheios-project/lexis-cs#8]
I am not sure why, but for some reason it looks like the _populatePermanentStorage promise never resolves on mobile (at least not on my Samsung Galaxy 7)

I was able to debug it as far as the definition of the Promise from _populatePermanentStorage, but from there it never seems to resolve and set the isReady flag to true.  No errors are logged to the console.

**Comment by kirlat** [comment]
Hmm, I can see dictionary records stored to the IndexedDB on a mobile device, but not the metadata.

**Comment by kirlat** [comment]
I was testing it on my phone (which is quite old but because of this is good for testing) and it seems that insertion of CEDICT dictionary data is very slow. It took 3 minutes to complete. Insertion of CEDICT metadata that started a moment earlier was never completed, however. That's why `_populatePermanentStorage` never resolves: it waits for both operations to complete.

I tried to remove the code that inserts a dictionary data and after that metadata was inserted successfully. So I think the problem is that a dictionary data insert request, which takes a lot of resources, blocks the metadata update request and the latter never completes.

I will try to execute those two requests sequentially and will check if it will fix the problem.

**Comment by balmas** [comment]
Ok, that makes sense. But it might be better to look at #11 before spending much time on this. I think we're going to revisit some of this design to have other fallback solutions to the local IndexedDB anyway.

**Comment by kirlat** [comment]
I've implemented a fix already: it was easy once the root cause is found. I deployed an updated code to AWS S3 bucket. It prints some messages that allow to see what is going on with storing the data. It will also allows to estimate duration of operations if timestamps are on in the console.

It took 20 seconds for me to store CEDICT records on desktop and 3 minutes on mobile. I'm wondering how is it for you.
